### PR TITLE
Integration of the Weights & Biases Metric Logger

### PR DESCRIPTION
**Pull request by manangoel99**
Based on #3358

This PR adds the WandbLogger callback to let users log their models and training/evaluation metrics to Weights & Biases. The callback logs the training metrics at every call of the `on_step_end` hook and logs the evaluation metrics along with model checkpoints tagged if they are they best at the `on_epoch_end` hook.

To use the W&B Logger, the users just need to add the command line argument `--use_wandb True` or add the `use_wandb: True` flag to the config.yml file. To add other arguments for logging metrics to W&B like project name, entity name, run name and so on the users can do that in two ways:

1. ```python tools/train -c config.yml --use_wandb -o wandb-project=MyDetector wandb-entity=MyTeam wandb-save_dir=./logs```
The arguments to the W&B logger must be proceeded by `-o` and each invidiual argument must contain the prefix "wandb-".

2. An alternative way is to add the arguments to the `config.yml` file under the `wandb` header. For example
```
use_wandb: True
wandb:
    project: MyProject
    entity: MyTeam
    save_dir: ./logs
```

This will generate a dashboard for the user with visualizations of the training and evaluation metrics. The following dashboard was made using the coco128 subset of COCO. 
Screenshot 2022-05-06 at 3 28 05 AM
 
Along with plots of the training and evaluation metrics, all the config variables are also logged to the dashboard.
Screenshot 2022-05-06 at 3 29 25 AM

The logged models are uploaded and tagged according to which epoch they were generated in and if it is the best model.
Screenshot 2022-05-06 at 3 30 41 AM

The sample workspace is available [here](https://wandb.ai/manan-goel/dcn)

### CODE REVIEW

File: docs/tutorials/logging_en.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+For example
+
+```
+python tools/train -c config.yml --use_vdl --vdl_log_dir ./logs
+```
+
+Another possible way to do this is to add the aforementioned flags to the `config.yml` file.
+
+## Weights & Biases
+W&B is a MLOps tool that can be used for experiment tracking, dataset/model versioning, visualizing results and collaborating with colleagues. A W&B logger is integrated directly into PaddleOCR and to use it, first you need to install the wandb sdk and login to your wandb account.
```
Comment (qingqing01) [inline docs/tutorials/logging_en.md]: PaddleOCR  -> PaddleDetection

**Review by qingqing01 — COMMENTED** [review]
Greaty job!

File: docs/tutorials/logging_en.md (hunk 1 above)
Comment (manangoel99) [reply docs/tutorials/logging_en.md]: Fixed! Apologies for the mistake😅

File: ppdet/engine/trainer.py [hunk 2]
```diff
@@ -182,6 +182,8 @@ def _init_callbacks(self):
                 self._callbacks.append(VisualDLWriter(self))
             if self.cfg.get('save_proposals', False):
                 self._callbacks.append(SniperProposalsGenerator(self))
+            if self.cfg.get('use_wandb', True) or 'wandb' in self.cfg:
```
Comment (wangxinxin08) [inline ppdet/engine/trainer.py]: Maybe replacing this line with the following code is better:
``` python
if self.cfg.get('use_wandb', False):
```

**Review by wangxinxin08 — APPROVED** [review]
LGTM

**Review by qingqing01 — APPROVED** [review]

**Comment by manangoel99** [comment]
Hi @wangxinxin08 ! Could you let me know when is the next release expected?

**Comment by wangxinxin08** [comment]
> Hi @wangxinxin08 ! Could you let me know when is the next release expected?

Do you mean the next release of PaddleDetection? Maybe it will be released in August

**Pull request merged**